src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,774 @@
+            col_ind_cos += [2*j-1, 2*j]
+
+            data_sin += [-1, 1]
+            row_ind_sin += [2*j-2*n, 2*j-2*n-1]
+            col_ind_sin += [2*j-1, 2*j]
+
+            if j + n <= self._num_freqs:
+
+                data_cos += [1, 1]
+                row_ind_cos += [2*j+2*n-1, 2*j+2*n]
+                col_ind_cos += [2*j-1, 2*j]
+
+                data_sin += [1, -1]
+                row_ind_sin += [2*j+2*n, 2*j+2*n-1]
+                col_ind_sin += [2*j-1, 2*j]
+
+        # Make sparse matrices
+        cos_matrix = sparse.csr_matrix(
+            (data_cos, (row_ind_cos, col_ind_cos)))
+        sin_matrix = sparse.csr_matrix(
+            (data_sin, (row_ind_sin, col_ind_sin)))
+
+        return cos_matrix, sin_matrix
+
+    def _vector_product(self, vectors, round_data):
+
+        '''
+        Calculates cos^2(n\phi/2+\beta/2)P(\phi) for an input
+        n, beta and P(\phi).
+
+        Args:
+            vectors: the input vectors to be multiplied
+
+            round_data: dictionary containing data about rotation
+                in particular, round data needs 'num_rotations',
+                'final_rotation', and 'measurement'.
+
+        Returns:
+            updated vectors: the vector representation
+                of P_{k,\beta}(m|\phi_j)P_{prior}(\phi_j) for each
+                input vector P_{prior}(\phi_j)
+        '''
+
+        # Extract the pieces we want from the round_data
+        beta = round_data['final_rotation'] -\
+            round_data['measurement'] * pi
+        n = round_data['num_rotations']
+
+        # Get the matrices for multiplication
+        if n <= self._max_n:
+            cos_matrix = self._matrices[n-1][0]
+            sin_matrix = self._matrices[n-1][1]
+        else:
+            cos_matrix, sin_matrix = self._make_matrix(n)
+
+        return 0.5*vectors +\
+            0.25*cos(beta)*cos_matrix.dot(vectors) +\
+            0.25*sin(beta)*sin_matrix.dot(vectors)
+
+    def _mlikelihood(self, a_vec):
+        # Calculates the negative likelihood of a set of amplitudes
+        # generating the observed measurements
+
+        return -sum([np.log(p_vec.dot(a_vec[:len(p_vec)]))
+                    for p_vec in self._p_vecs]) +\
+            self._init_mlikelihood(a_vec)
+
+    def _diff_mlikelihood(self, a_vec):
+        # The derivative of the above function
+        return -sum([p_vec/(p_vec.dot(a_vec[:len(p_vec)]))
+                    for p_vec in self._p_vecs]) +\
+            self._dinit_mlikelihood(a_vec)
+
+    def _single_diff(self, p_vec, a_vec):
+        # The first derivative of a single term in the
+        # likelihood function.
+        return -p_vec / np.dot(p_vec, a_vec[:len(p_vec)])
+
+    def _jacobian_term(self, p_vec, a_vec):
+        # The second derivative of the above function,
+        # evaluated for a single p vector - to evaluate
+        # and store
+        return np.dot(p_vec[:, np.newaxis], p_vec[np.newaxis, :]) /\
```

Comment:
this method of breaking lines is definitely going to cause problems with the linter. We really should get automatic linting set up... please just lint the entire PR.

Reply:
Odd, it hasn't done it for me. Would you prefer that I bracket everything?